data_extrator.py

```diff
@@ -22,14 +18,26 @@ def exponential_filter(time_step):
         return filtered_data[-parameters_per_slice:]
 
 
-    parameters = np.zeros((slice_count, parameters_per_slice))
-    for i in range(slice_count):
-        # Calculate last point of slice
-        time_step = time_step_0 - i*slice_shift
-        # Calculate last 6 points of filtered values
-        parameters[i] = exponential_filter(time_step)
+    def exponential_filter_parameters(data, time_step_0, slice_count=5, slice_overlap=50, alpha_value = 0.8, slice_size = 100, parameters_per_slice = 6, plot_results = False):
+        # Check if enough data points
+        slice_shift = slice_size - slice_overlap
+        if time_step_0 - (slice_count-1) * slice_shift - slice_size < 0:
```

Comment:
W tym wzorze jest coś nie tak. Zakładając slice_count = 5, slice_size = 20 i slice_overlap = 0 mamy, że time_step_0 musi być mniejsze niż (5-1)*20-20=60? Chyba musi być większe niż ta liczba plus time_step_0 w tym przypadku musi być conajmniej 100?
Co do wyliczania tej minimalnej wartości, lepiej będzie zrobić do tego funkcję, która zwraca minimalny time_step_0, żeby potem dało jej się użyć przy ewaluacji (żeby w niej zacząć liczenie od tego kroku).

Reply:
`- (slice_count-1) * slice_shift - slice_size = -(5-1)*20-20 = -100 -> time_step_zero - 100 > 0`, zgubiłeś minus we wzorze, z nim się zgadza

Dodam funkcję od minimalnego kroku o której piszesz.

Reply:
Teraz widzę, wygląda ok